### DISCUSSION TITLE
Can process pod declaration merge with config pod declaration?

### DISCUSSION THREAD

**Discussion by alanbchristie** [nextflow-io/nextflow#2933] [Q&A]
I have a `nextflow.config` with a process **pod** declaration, i.e. something like this...

```
process {
  pod = [ [nodeSelector: 'example.com/purpose=core'],
          [label: 'example.com/instance-id',
           value: 'instance-7be5e19c-02c9-427f-a840-581728d46397'] ]
}
```

As you can see, it defines a `nodeSelector` and a `label`.

But, if I run a process that also defines a **pod** declaration, with different (non-overlapping) properties, like this...

```
process combine {

    pod imagePullSecret: 'example-com-secret'
    container 'registry.gitlab.com/example/example:stable'

    [...]
}
```

...the **pod** declaration in the `nextflow.config` appears to be lost - i.e. the `nodeSelector` and `labels` are not applied to the process **pod**.

So...

1. Is there a way to combine (merge) the pod declarations?
2. ...if not, would it actually be useful to others? i.e. does this warrant a bug/feature request?
3. ...if not, can the documentation be adjusted to make it clear that in-process declarations replace configured declarations with a clear example?

**Comment by pditommaso** [comment]
This is the expected behaviour, likely it should be documented better. 

Regarding this specific use case, we are working on improving the [support for labels](https://github.com/nextflow-io/nextflow/issues/2845). We'll take in consideration your point.

**Reply by alanbchristie** [reply]
Thank you. The ability to _merge_ **pod** declarations with the config would provide an extremely flexible way to configure the workflow.

For our specific case, as we are simply setting a Kubernetes `imagePullSecret`, I realise that we can avoid the config **pod** declaration from being lost by attaching the secret(s) to the `ServiceAccount` rather than via the **pod**. That way the config **pod** declaration is not lost.
